src/components/discover-sheet/DiscoverSearch.js

```diff
@@ -0,0 +1,252 @@
+import { map, toLower } from 'lodash';
+import matchSorter from 'match-sorter';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { IS_TESTING } from 'react-native-dotenv';
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import { useDispatch } from 'react-redux';
+import { addHexPrefix } from '../../handlers/web3';
+import CurrencySelectionTypes from '../../helpers/currencySelectionTypes';
+import tokenSectionTypes from '../../helpers/tokenSectionTypes';
+import { emitAssetRequest } from '../../redux/explorer';
+import deviceUtils from '../../utils/deviceUtils';
+import { CurrencySelectionList } from '../exchange';
+import { initialChartExpandedStateSheetHeight } from '../expanded-state/ChartExpandedState';
+import { Row } from '../layout';
+import DiscoverSheetContext from './DiscoverSheetContext';
+import {
+  useAccountAssets,
+  useTimeout,
+  useUniswapAssets,
+  useUniswapAssetsInWallet,
+} from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import Routes from '@rainbow-me/routes';
+import { filterList } from '@rainbow-me/utils';
+
+const headerlessSection = data => [{ data, title: '' }];
+
+const searchCurrencyList = (searchList = [], query) => {
+  const isAddress = query.match(/^(0x)?[0-9a-fA-F]{40}$/);
+
+  if (isAddress) {
+    const formattedQuery = toLower(addHexPrefix(query));
+    return filterList(searchList, formattedQuery, ['address'], {
+      threshold: matchSorter.rankings.CASE_SENSITIVE_EQUAL,
+    });
+  }
+
+  return filterList(searchList, query, ['symbol', 'name'], {
+    threshold: matchSorter.rankings.CONTAINS,
+  });
+};
+
+const timingConfig = { duration: 700 };
+
+export default function DiscoverSearch() {
+  const { navigate } = useNavigation();
+  const listOpacity = useSharedValue(0);
+  const { allAssets } = useAccountAssets();
+
+  const listAnimatedStyles = useAnimatedStyle(() => {
+    return {
+      opacity: listOpacity.value,
+    };
+  });
+
+  useEffect(() => {
+    listOpacity.value = withTiming(1, timingConfig);
+  }, [listOpacity]);
+
+  const { setIsSearching, searchQuery, isSearchModeEnabled } = useContext(
+    DiscoverSheetContext
+  );
+  const [searchQueryForSearch, setSearchQueryForSearch] = useState('');
+  const type = CurrencySelectionTypes.output;
+  const dispatch = useDispatch();
+  const {
+    curatedNotFavorited,
+    favorites,
+    globalHighLiquidityAssets,
+    globalLowLiquidityAssets,
+    globalVerifiedAssets,
+    loadingAllTokens,
+  } = useUniswapAssets();
+  const { uniswapAssetsInWallet } = useUniswapAssetsInWallet();
+  const { colors } = useTheme();
+
+  const currencyList = useMemo(() => {
```

Comment:
Can be done later: but let's pull out this section of search logic as a hook as it is exactly the same as the one in CurrencySelectModal (there may be some trickiness with typing, but it'll be good to do)

Reply:
Sounds good. I can tackle after releasing.